src/Microsoft.VisualStudio.Threading/JoinableTaskContext.cs

```diff
@@ -339,6 +339,27 @@ public bool IsMainThreadBlocked()
             return false;
         }
 
+        /// <summary>
+        /// Gets a very likely value whether the main thread is blocked for the caller's completion.
+        /// It is less accurate when the UI thread blocking task just starts and hasn't been blocked yet, or the dependency chain is just removed.
+        /// However, unlike <see cref="IsMainThreadBlocked"/>, this implementation is lock free, and faster in high contention scenarios.
```

Comment:
From the name and docs it seems the risk is a false positive and never a false negative. Is that right? It may be worth calling this out explicitly in the docs.

Reply:
It can be false negative.  There is a sample scenario it can be false negative:

The UI thread is inside a JTF.Run, it RunAsync to spin off a child task, and will wait on that. Inside the child task, if it checks IsMainThreadBlocked immediately (the previous function), it will return true. But the new implementation will return false. The reason is that the UI thread task has never been blocked once. (The new method will return true inside the JTF.Run task itself under this condition, but not in its dependent tasks).

The reason is that we optimized the code to make small non blocking method wrapped inside JTF.Run function to run faster without establishing extra data structure. The original function walks dependencies graph in that condition, which turns out to be expensive.